google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/ManagedCloudSdkService.java

```diff
@@ -248,4 +234,66 @@ private void updateStatus(SdkStatus sdkStatus) {
   private void notifyListeners(CloudSdkService sdkService, SdkStatus status) {
     statusUpdateListeners.forEach(listener -> listener.onSdkStatusChange(sdkService, status));
   }
+
+  enum ManagedSdkJobType {
+    INSTALL,
+    UPDATE
+  }
+
+  /**
+   * Managed SDK Job future listener, handles success/error logic, logs errors, shows notifications
+   * to a user, updates SDK service statuses.
+   */
+  private final class ManagedSdkJobListener implements FutureCallback<Path> {
+    private final ManagedSdkJobType jobType;
+
+    private ManagedSdkJobListener(ManagedSdkJobType jobType) {
+      this.jobType = jobType;
+    }
+
+    @Override
+    public void onSuccess(Path path) {
+      logger.info("Managed Google Cloud SDK successfully installed/updated at: " + path);
+
+      updateStatus(SdkStatus.READY);
+
+      ManagedCloudSdkServiceUiPresenter.getInstance().notifyManagedSdkJobSuccess(jobType);
+    }
+
+    @Override
+    public void onFailure(Throwable t) {
+      if (t instanceof InterruptedException) {
+        logger.info("Managed Google Cloud SDK install/update cancelled.");
+
+        ManagedCloudSdkServiceUiPresenter.getInstance().notifyManagedSdkJobCancellation(jobType);
+      } else {
+        logger.warn("Error while installing/updating managed Cloud SDK", t);
+
+        ManagedCloudSdkServiceUiPresenter.getInstance()
+            .notifyManagedSdkJobFailure(jobType, t.toString());
```

Comment:
I think this makes sense to visibly show the user the throwable message here right? Otherwise it would be hard to diagnose without opening the logs. I'm going back and forth on this since I have a similar dilemma in my other PR.

Reply:
Exactly, plus hopefully we will not hit these errors frequently on major platforms, and exception messages are pretty clear.